bk-api/bk_api.py

```diff
@@ -1316,102 +1253,95 @@ def get(self):
             # first check the default beehive exists
             try:
                 bee_db = BeekeeperDB()
-                beehive_obj = bee_db.get_beehive(DEFAULT_BEEHIVE)
+                beehive_obj = bee_db.get_beehive(beehive_id)
                 bee_db.close()
             except Exception as e:
-                raise Exception(f"error: get_beehive returned: {str(e)}")
+                raise Exception(f"error: get_beehive returned: {e}")
 
             if not beehive_obj:
-                raise Exception(f"error: Beehive {DEFAULT_BEEHIVE} is not known yet" )
+                logger.error("registration error: beehive does not exist: %r", request.args)
+                return "error: beehive not found", HTTPStatus.NOT_FOUND
 
         try:
             # create keypair and certificates for node (idempotent function)
-            registration_result =  _register(node_id)
+            registration_result = _register(node_id)
         except Exception as e:
-            logger.debug(f"_register failed: {str(e)}")
-            traceback.print_exc()
-            logger.debug(f"Error: unable to register id [{node_id} , {str(e)}]")
-            return f"Error: unable to register id [{node_id} , {str(e)}]\n", 500
+            logger.exception("registration error: _register: %r", request.args)
+            return f"error: unable to register id [{node_id} , {e}]\n", HTTPStatus.INTERNAL_SERVER_ERROR
 
         # update beekeeper db (create registartion event)
         try:
             register_node(node_id, lock_requested_by="register-resource")
         except Exception as e:
-            logger.debug(f"Error: Creating registration event failed: {str(e)}")
-            return f"Error: Creating registration event failed: {str(e)}", 500
+            logger.exception("registration error: register_node: %r", request.args)
+            return f"Error: Creating registration event failed: {str(e)}", HTTPStatus.INTERNAL_SERVER_ERROR
 
-
-        if DEFAULT_BEEHIVE:
+        if beehive_id:
             time.sleep(2) # see if that helps with the locks
-            logger.debug("Adding user [{}] to default beehive".format(node_id))
+            # TODO(sean) Review how locks are used. I'm concerned that a sleep is supposed to "help with the locks".
             try:
-                set_node_beehive(node_id, DEFAULT_BEEHIVE)
+                set_node_beehive(node_id, beehive_id)
             except Exception as e:
-                logger.debug(f"Error: Adding node to beehive {DEFAULT_BEEHIVE} failed: {str(e)}")
-                # Do not let registration fail because of this
-                return f"Error: Adding node to beehive {DEFAULT_BEEHIVE} failed: {str(e)}", 500
-        else:
-            logger.debug("No default beehive defined")
-
+                logger.exception("registration error: set_node_beehive: %r", request.args)
+                return f"Error: Adding node to beehive {beehive_id} failed: {e}", HTTPStatus.INTERNAL_SERVER_ERROR
```

Comment:
the comment that was here before 

```
# Do not let registration fail because of this
```

was this not true?  did we fail registration if the beehive assignment didn't work?

Reply:
better question, what do we _want_ to do? if someone attempts registration for an invalid beehive, should we reject registration?  (i sorta think, 'yes`).

Reply:
After looking at that again, I think that comment may be misleading or not worded intuitively.

Looking at the logic then (and now), we have:

```python
# before
try:
    set_node_beehive(node_id, DEFAULT_BEEHIVE)
except Exception as e:
    logger.debug(f"Error: Adding node to beehive {DEFAULT_BEEHIVE} failed: {str(e)}")
    # Do not let registration fail because of this
    return f"Error: Adding node to beehive {DEFAULT_BEEHIVE} failed: {str(e)}", 500

# after (no functional changes)
try:
    set_node_beehive(node_id, beehive_id)
except Exception as e:
    logger.exception("registration error: set_node_beehive: %r", request.args)
    return f"Error: Adding node to beehive {beehive_id} failed: {e}", HTTPStatus.INTERNAL_SERVER_ERROR
```

So, it looks like BK _does_ indicate that registration failed if we can't set_node_beehive.

Reply:
I also agree, I think this is the right behavior. I think allowing this to pass and telling the node "all clear" when there's some internal inconsistency would be the wrong thing to do.